x-pack/plugins/lens/public/indexpattern_plugin/indexpattern.test.tsx

```diff
@@ -248,11 +263,184 @@ describe('IndexPattern Data Source', () => {
+        });
+
+        it('is droppable if the field is supported by filterOperations', () => {
+          const component = shallow(
+            <IndexPatternDimensionPanel
+              dragDropContext={{
+                ...dragDropContext,
+                dragging: { type: 'number', name: 'bar' },
+              }}
+              state={dragDropState()}
+              setState={() => {}}
+              columnId={'col2'}
+              filterOperations={op => op.dataType === 'number'}
+            />
+          );
+
+          const dropTarget = component
+            .find('[data-test-subj="indexPattern-dropTarget"]')
+            .first()
+            .props();
+
+          expect((dropTarget as { droppable: boolean }).droppable).toBeTruthy();
+        });
+
+        it('appends the dropped column when a field is dropped', () => {
+          const dragging = { type: 'number', name: 'bar' };
+          const testState = dragDropState();
+          const setState = jest.fn();
+          const component = shallow(
+            <IndexPatternDimensionPanel
+              dragDropContext={{
+                ...dragDropContext,
+                dragging,
+              }}
+              state={testState}
+              setState={setState}
+              columnId={'col2'}
+              filterOperations={op => op.dataType === 'number'}
+            />
+          );
+
+          const dropTarget = component
+            .find('[data-test-subj="indexPattern-dropTarget"]')
+            .first()
+            .props();
+
+          const onDrop = (dropTarget as { onDrop: DropHandler }).onDrop;
+
+          onDrop(dragging);
+
+          expect(setState).toBeCalledTimes(1);
+          expect(setState.mock.calls).toMatchObject([
+            [
+              {
+                columns: {
+                  ...testState.columns,
+                  col2: {
+                    dataType: 'number',
+                    sourceField: 'bar',
+                  },
+                },
+              },
+            ],
+          ]);
+        });
+
+        it('ignores drops of unsupported fields', () => {
+          const dragging = { type: 'number', name: 'baz' };
+          const testState = dragDropState();
+          const setState = jest.fn();
+          const component = shallow(
+            <IndexPatternDimensionPanel
+              dragDropContext={{
+                ...dragDropContext,
+                dragging,
+              }}
+              state={testState}
+              setState={setState}
+              columnId={'col2'}
+              filterOperations={op => op.dataType === 'number'}
+            />
+          );
+
+          const dropTarget = component
+            .find('[data-test-subj="indexPattern-dropTarget"]')
+            .first()
+            .props();
+
+          const onDrop = (dropTarget as { onDrop: DropHandler }).onDrop;
+
+          onDrop(dragging);
+
+          expect(setState).not.toBeCalled();
+        });
+      });
```

Comment:
I think there is a test that is missing, since all of these tests were for a previously-empty target. Please add a test that you can replace the existing column.